### DISCUSSION TITLE
AttributeError in Lesson 6 - 5:36:33 timestamp

### DISCUSSION THREAD

**Discussion by ArcusConsulting** [smartcontractkit/full-blockchain-solidity-course-py#323] [Q&A] [answered]
Not sure why I'm getting an AttributeError:
![image](https://user-images.githubusercontent.com/91853555/140662910-a70aba90-d739-4413-9748-370e212fe961.png)

![image](https://user-images.githubusercontent.com/91853555/140662869-44d38637-edff-4b9a-a520-bb64fcd53f12.png)

![image](https://user-images.githubusercontent.com/91853555/140662889-d7133279-3b14-4e03-a494-a3668f74992a.png)

**Comment by cromewar** [comment]
1. Make sure you have the MockV3Aggregator.sol on contracts/tests-
2. Delete your build folder and compile again `brownie compile`.

Hopes this help.

**Reply by ArcusConsulting** [reply]
MockV3Aggregator.sol is in contracts/test, and I deleted my build folder and compiled again... Still getting an error.

**Reply by PatrickAlphaC** [reply]
What's your `MockV3Aggregator.sol` look like?

**Reply by cromewar** [reply]
I see, please as @PatrickAlphaC  suggested could you share your MockV3Aggregator code here please? or copy-past it from here: https://github.com/PatrickAlphaC/brownie_fund_me/blob/main/contracts/test/MockV3Aggregator.sol

**Reply by ArcusConsulting** [reply]
Copy-pasted from GitHub... See below

**Reply by cromewar** [reply]
It seems everything is ok, for me is a problem with your brownie installation, using the recommendations on this thread: https://github.com/smartcontractkit/full-blockchain-solidity-course-py/discussions/295

- use virtual env,
- install brownie to the venv,
- reinstall your ganache-cli to the directory you own (or simply chown your node_modules dir)

**Reply by ArcusConsulting** [reply]
... I don't know how to use virtual env and i don't know what you mean in your 3rd bullet point. I'll see what i can manage later tonight after work.

**Reply by cromewar** [reply]
[Here](https://packaging.python.org/guides/installing-using-pip-and-virtual-environments/) you have a guide about python virtual envs.
The third bullet point means to not install ganache using super user privileges (like using admin on windows o Sudo on linux).

**Comment by ArcusConsulting** [comment]
![image](https://user-images.githubusercontent.com/91853555/140664052-bbca8f9c-4e3b-40a4-9405-8ed475ec70ac.png)

**Comment by ArcusConsulting** [comment]
Still getting the same error:
![image](https://user-images.githubusercontent.com/91853555/140664087-ee26ba7d-b314-4744-8457-b778da3e9263.png)

**Comment by ArcusConsulting** [comment]
`// SPDX-License-Identifier: MIT

pragma solidity ^0.6.0;

import "@chainlink/contracts/src/v0.6/interfaces/AggregatorV2V3Interface.sol";

/**
 * @title MockV3Aggregator
 * @notice Based on the FluxAggregator contract
 * @notice Use this contract when you need to test
 * other contract's ability to read data from an
 * aggregator contract, but how the aggregator got
 * its answer is unimportant
 */
contract MockV3Aggregator is AggregatorV2V3Interface {
    uint256 public constant override version = 0;

    uint8 public override decimals;
    int256 public override latestAnswer;
    uint256 public override latestTimestamp;
    uint256 public override latestRound;

    mapping(uint256 => int256) public override getAnswer;
    mapping(uint256 => uint256) public override getTimestamp;
    mapping(uint256 => uint256) private getStartedAt;

    constructor(uint8 _decimals, int256 _initialAnswer) public {
        decimals = _decimals;
        updateAnswer(_initialAnswer);
    }

    function updateAnswer(int256 _answer) public {
        latestAnswer = _answer;
        latestTimestamp = block.timestamp;
        latestRound++;
        getAnswer[latestRound] = _answer;
        getTimestamp[latestRound] = block.timestamp;
        getStartedAt[latestRound] = block.timestamp;
    }

    function updateRoundData(
        uint80 _roundId,
        int256 _answer,
        uint256 _timestamp,
        uint256 _startedAt
    ) public {
        latestRound = _roundId;
        latestAnswer = _answer;
        latestTimestamp = _timestamp;
        getAnswer[latestRound] = _answer;
        getTimestamp[latestRound] = _timestamp;
        getStartedAt[latestRound] = _startedAt;
    }

    function getRoundData(uint80 _roundId)
        external
        view
        override
        returns (
            uint80 roundId,
            int256 answer,
            uint256 startedAt,
            uint256 updatedAt,
            uint80 answeredInRound
        )
    {
        return (
            _roundId,
            getAnswer[_roundId],
            getStartedAt[_roundId],
            getTimestamp[_roundId],
            _roundId
        );
    }

    function latestRoundData()
        external
        view
        override
        returns (
            uint80 roundId,
            int256 answer,
            uint256 startedAt,
            uint256 updatedAt,
            uint80 answeredInRound
        )
    {
        return (
            uint80(latestRound),
            getAnswer[latestRound],
            getStartedAt[latestRound],
            getTimestamp[latestRound],
            uint80(latestRound)
        );
    }

    function description() external view override returns (string memory) {
        return "v0.6/tests/MockV3Aggregator.sol";
    }
}

// MockOracle
// Function signatures, event signatures, log topics`

**Comment by guillermo-palou** [comment, marked as answer]
I had the same error! Remember to put the parentheses after get_account in the MockV3Aggregator.deploy

        MockV3Aggregator.deploy(
            DECIMALS, Web3.toWei(STARTING_PRICE, "ether"), {"from": get_account()}

**Reply by cromewar** [reply]
@ArcusConsulting does this work for you?

**Reply by ArcusConsulting** [reply]
Will let you know tonight!

**Reply by ArcusConsulting** [reply]
It worked! Thank you, everyone!

**Reply by PatrickAlphaC** [reply]
Wooooo!!!

**Reply by DarrenWayn** [reply]
the get_account => () don't forget to use this as a function
